lang/src/test/kotlin/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransformTests.kt

```diff
@@ -877,20 +877,21 @@ class AstToLogicalVisitorTransformTests {
         )
     }
 
-    data class ProblemTestCase(val sql: String, val expectedProblem: Problem)
+    data class ProblemTestCase(val id: Int, val sql: String, val expectedProblem: Problem)
 
     @ParameterizedTest
     @ArgumentsSource(ArgumentsForProblemTests::class)
     fun `unimplemented features are blocked`(tc: ProblemTestCase) {
+        val id by lazy { "[in test case #${tc.id}]" }
         val problemHandler = ProblemCollector()
-        assertDoesNotThrow("Parsing TestCase.sql should not throw") {
+        assertDoesNotThrow("Parsing TestCase.sql should not throw $id") {
```

Comment:
Instead of adding _id_ couldn't we use `ProblemTestCase`'s _val_ using id in such a way seems odd IMO.

Reply:
Did you mean using `ProblemTestCase`'s `sql` field? 

My purpose in introducing `id` was making it possible to point at the offending `ProblemTestCase` when the test fails.  The assert message as it was, was not informative. The `sql` field could work, but its contents are longer and there could be instances with only subtle differences, complicating the task of determining the correct one when fixing the tests.

Reply:
Makes sense, but wouldn't having a free flow `id` cause other problems; since we don't auto-generate or check for uniqueness?

Reply:
Yes, still not ideal. But we are in the context of comparing three viable alternatives that are there on hand for identifying the offending test: (1) leave it as it was, no identification; (2) identify with the full text of the query; (3) identify with a short integer.
This is internal testing; the driving principle for choosing among the 3 is efficiency of identifying the broken test that needs fixing. Meta-F on an integer beats the alternatives in my mind.

Reply:
I guess using `id` is what made me think it's odd. It could be just a name for the test case; happy to park the discussion here though.